Plugins/Public/base_plugin/PlayerCommands.cpp

```diff
@@ -1736,6 +1742,65 @@ namespace PlayerCommands
 		pub::Player::PopUpDialog(client, caption, message, POPUPDIALOG_BUTTONS_CENTER_OK);
 	}
 
+	//FL only supports one "price" in the dealer menu, this function gives players a command to 
+	//switch the view between the bases "buy" and "sell" price
+	void PriceView(uint client, const wstring& args)
+	{
+		// Check that this player is in a player controlled base
+		PlayerBase* base = GetPlayerBaseForClient(client);
+		if (!base)
+		{
+			PrintUserCmdText(client, L"ERR Not in player base");
+			return;
+		}
+		SendResetMarketOverride(client);
+		const wstring& cmd = GetParam(args, ' ', 1);
+		// Send the market
+		if (cmd == L"buy")
+		{
+			for (map<uint, MARKET_ITEM>::iterator i = base->market_items.begin();
+				i != base->market_items.end(); i++)
+			{
+				uint good = i->first;
+				MARKET_ITEM& item = i->second;
+				wchar_t buf[200];
+
+				//this if statement prevents items intended to be hidden from being displayed
+				//when the /pice commands run
+				if (item.min_stock > 2000 || (item.min_stock == item.max_stock)) {

```

Comment:
Why this condition to hide if min_stock > 2000?

Reply:
This was in attempt to invis stock that hasn't been configured in /shop price menu and from looking at some examples of live bases, to handle coniditions where owners use arbitrary high number to hide supply stock

Reply:
If I remember correctly the default behaviour:
The players can see the prices of items which player base is willing to sell (if item.count > min_stock), or willing to buy (if item.count < max_stock)
Plus player can see all shop prices if he has guest base password.

What's exactly wrong with the current permission system, and why not to use it again? Why it needs to have item invisibility added to it with additional conditions?

Reply:
The player hide the stock of item with ridiculous high min_stock value, you are right.
And lets say it is the base with Ore. Which can have up to... 100 or 200k thousands of Ores.
So.... the min_stock > 2000 would be not really useful to hide it at all.

High min_stock forbids from selling (if min_stock is higher than item count, the base will not sell anything)
Zero max_stock forbids from buying. (if max_stock 0, the base will not buy anyithing)

What does min_stock == max_stock do then?

Reply:
I'll start by answering the easy part: min_stock == max_stock condition is there because by default, items are loaded into the /shop as min=100000 max=100000, so setting that condition will auto hide those items until owner/manager configures what he wants them to be. Also to disable buy/sell, currently players can set min and max to 0, so it takes care of keeping these items invis'ed again.

upon looking further, the min_stock > 2000 does seem redundant and not of use, though I thought I had a reason when configuring it. (need to keep better comments it seems).

Reply:
> Also to disable buy/sell, currently players can set min and max to 0, so it takes care of keeping these items invis'ed again.

I will repeat again. When player sets min and max to 0. He sets to sell all of his items. It is perfectly valid selling mod for the item.

Reply:
yes you're right, it does allow selling of remaining stock, in current set up, owners keep items in this state and sell ore to the base while logged in to override these values and not have to switch them, essentially keeping base in sell mode. But that is based on the limitation of the current set up. With the ability to have separate buy and sell prices, base owners will only use this min 0, max 0 if they want to shut down both buying and selling and therefore won't need to show their prices.

As stated this conditional is there because with the /price command it reveals all set prices of everything listed in /shop, but with the conditional in place, it only shows relevant values.